### PR TITLE
fix: add more kubernetesConfig nil checks to account for more resilience

### PR DESCRIPTION
**Pull request by tariq1890**
More nil checks! One can never be too safe!

### CODE REVIEW

File: pkg/engine/masterarmresources.go [hunk 1]
```diff
@@ -52,14 +57,18 @@ func createKubernetesMasterResourcesVMAS(cs *api.ContainerService) []interface{}
...
+		}
 
 		if provisionJumpbox {
 			jumpboxVM := createJumpboxVirtualMachine(cs)
 			masterResources = append(masterResources, jumpboxVM)
 			jumpboxIsManagedDisks :=
-				cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() &&
-					cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
+				kubernetesConfig.PrivateJumpboxProvision() &&
+					kubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
```
Comment (jackfrancis) [inline pkg/engine/masterarmresources.go]: do we know that `JumpboxProfile` will never be nil here?

File: pkg/engine/virtualmachines.go [hunk 2]
```diff
@@ -220,12 +226,15 @@ func createJumpboxVirtualMachine(cs *api.ContainerService) VirtualMachineARM {
 		DataDisks: &[]compute.DataDisk{},
 	}
 
-	jumpBoxIsManagedDisks := cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() && cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
+	var jumpBoxIsManagedDisks bool
+	if kubernetesConfig != nil && kubernetesConfig.PrivateCluster != nil {
+		jumpBoxIsManagedDisks = kubernetesConfig.PrivateJumpboxProvision() && kubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
```
Comment (jackfrancis) [inline pkg/engine/virtualmachines.go]: ditto here

File: pkg/engine/masterarmresources.go (hunk 1 above)
Comment (tariq1890) [reply pkg/engine/masterarmresources.go]: The `PrivateJumpboxProvision()` function checks for `JumpoboxProfile`. So we are covered that way.

File: pkg/engine/virtualmachines.go (hunk 2 above)
Comment (tariq1890) [reply pkg/engine/virtualmachines.go]: same answer as above.

File: pkg/engine/virtualmachines.go [hunk 3]
```diff
@@ -220,12 +226,15 @@ func createJumpboxVirtualMachine(cs *api.ContainerService) VirtualMachineARM {
...
+	}
 
 	if jumpBoxIsManagedDisks {
 		storageProfile.OsDisk = &compute.OSDisk{
 			CreateOption: compute.DiskCreateOptionTypesFromImage,
-			DiskSizeGB:   to.Int32Ptr(int32(cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.OSDiskSizeGB)),
+			DiskSizeGB:   to.Int32Ptr(int32(kubernetesConfig.PrivateCluster.JumpboxProfile.OSDiskSizeGB)),
```
Comment (jackfrancis) [inline pkg/engine/virtualmachines.go]: ditto

**Comment by jackfrancis** [comment]
/lgtm

pending unit tests

**Pull request merged**